src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs

```diff
@@ -16,23 +19,55 @@ public override IEnumerable<TResult> Select<TResult>(Func<int, TResult> selector
 
             public int[] ToArray()
             {
-                int[] array = new int[_end - _start];
-                Fill(array, _start);
+                int start = _start;
+                int[] array = new int[_end - start];
+                Fill(array, start);
                 return array;
             }
 
             public List<int> ToList()
             {
-                List<int> list = new List<int>(_end - _start);
-                Fill(SetCountAndGetSpan(list, _end - _start), _start);
+                int start = _start;
+                int end = _end;
+                List<int> list = new List<int>(end - start);
+                Fill(SetCountAndGetSpan(list, end - start), start);
                 return list;
             }
 
-            private static void Fill(Span<int> destination, int value)
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static void Fill(Span<int> destination, int start)
```

Comment:
why is this better than just Span.Fill ?

Reply:
This one is for filling in sequences of e.g. 0, 1, 2, 3, 4, ...N.
Or if you mean why the inline was added - previously `Fill` was getting inlined in Tier 1 and the attribute gives extra help to ensure short ranges do not regress. This is okay here because the caller itself won't get inlined so it only affects the bodies of `ToArray()` and `ToList()`.

Reply:
Ah I see, I confused it with "fill with value"